pkg/controllers/managementuserlegacy/helm/controller.go

```diff
@@ -269,29 +269,31 @@ func (l *Lifecycle) Remove(obj *v3.App) (runtime.Object, error) {
 			return obj, err
 		}
 	}
-	tempDirs, err := createTempDir(obj)
-	if err != nil {
-		return obj, err
-	}
-	defer os.RemoveAll(tempDirs.FullPath)
-	tokenName, err := l.writeKubeConfig(obj, tempDirs.KubeConfigFull, true)
-	if err != nil {
-		return obj, err
-	}
-	defer func() {
-		if err := l.systemTokens.DeleteToken(tokenName); err != nil {
-			logrus.Errorf("cleanup for helm token [%s] failed, will not retry: %v", tokenName, err)
+	if !(obj.Annotations["cattle.io/skipUninstall"] == "true") {
```

Comment:
A couple of  questions here:
1. In which case is this annotation used? In other words,  in which case that the uninstallation should be skipped?  
1. Is this annotation a new or existing one? 
2. Does UI add this annotation to the app?

Reply:
This is used after an app has been fully migrated and a user would like to deleted the old norman app object. The annotation is new. No it is meant to be added manually for now.

Reply:
ok, it sounds like something we need to add to the Rancher document and release notes.